internal/postgres/data_test.go

```diff
@@ -0,0 +1,237 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"fmt"
+	"harbourbridge/spanner/ddl"
+	"math/bits"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/civil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertData(t *testing.T) {
+	singleColTests := []struct {
+		name    string
+		t       ddl.ScalarType
+		isArray bool
+		pgt     string      // Postgres type (used by e.g. timestamp conversions).
+		in      string      // Input value for conversion.
+		e       interface{} // Expected result.
+	}{
+		{"bool", ddl.Bool{}, false, "", "true", true},
+		{"bytes", ddl.Bytes{ddl.MaxLength{}}, false, "", `\\x0001beef`, []byte{0x0, 0x1, 0xbe, 0xef}},
+		{"date", ddl.Date{}, false, "", "2019-10-29", getDate("2019-10-29")},
+		{"float64", ddl.Float64{}, false, "", "42.6", float64(42.6)},
+		{"int64", ddl.Int64{}, false, "", "42", int64(42)},
+		{"string", ddl.String{ddl.MaxLength{}}, false, "", "eh", "eh"},
+		{"timestamptz", ddl.Timestamp{}, false, "timestamptz", "2019-10-29 05:30:00+10", getTime(t, "2019-10-29T05:30:00+10:00")},
+		{"timestamp", ddl.Timestamp{}, false, "timestamp", "2019-10-29 05:30:00", getTime(t, "2019-10-29T05:30:00Z")},
+		// Add cases for each array type, since each is a separate code path.
+		{"bool array", ddl.Bool{}, true, "", "{true,false}", []bool{true, false}},
+		{"bytes array", ddl.Bytes{ddl.MaxLength{}}, true, "", `{\\x0001beef}`, [][]byte{{0x0, 0x1, 0xbe, 0xef}}},
+		{"date array", ddl.Date{}, true, "", "{2019-10-29,2019-10-28}", []civil.Date{getDate("2019-10-29"), getDate("2019-10-28")}},
+		{"float64 array", ddl.Float64{}, true, "", "{1.1,2.2,3.3}", []float64{1.1, 2.2, 3.3}},
+		{"int64 array", ddl.Int64{}, true, "", "{1,2,3}", []int64{1, 2, 3}},
+		{"string array", ddl.String{ddl.MaxLength{}}, true, "", "{1,2,3}", []string{"1", "2", "3"}},
+		{"timestamp array", ddl.Timestamp{}, true, "timestamptz", "{2019-10-29 05:30:00+10}", []time.Time{getTime(t, "2019-10-29T05:30:00+10:00")}},
+	}
+	for _, tc := range singleColTests {
+		table := "testtable"
+		col := "a"
+		conv := buildConv(table,
+			ddl.CreateTable{table, []string{col}, map[string]ddl.ColumnDef{col: ddl.ColumnDef{col, tc.t, tc.isArray, false, ""}}, []ddl.IndexKey{}, ""},
+			pgTableDef{map[string]pgColDef{col: pgColDef{id: tc.pgt}}})
+		conv.SetLocation(time.UTC)
+		ac, av, err := ConvertData(conv, table, table, []string{col}, []string{tc.in})
+		checkResults(t, ac, av, err, []string{col}, []interface{}{tc.e}, tc.name)
+	}
+
+	timestampTests := []struct {
+		name string
+		pgt  string
+		in   string
+		e    interface{}
+	}{
+		{"timestamptz hour timezone", "timestamptz", "2019-10-29 05:30:00+10:00", getTime(t, "2019-10-29T05:30:00+10:00")},
+		{"timestamptz hour/min timezone", "timestamptz", "2019-10-29 05:30:00+10:30", getTime(t, "2019-10-29T05:30:00+10:30")},
+		{"timestamptz no timezone", "timestamptz", "2019-10-29 05:30:00", getTime(t, "2019-10-29T05:30:00+11:00")},
```

Comment:
will this fail if I run the test in PST because it is taking local time?

Reply:
No. We explicitly set the location in the test body to ensure that the test is robust: see conv.SetLocation(loc) a few lines below (line 83).

Added a comment to clarify this.

Reply:
Ahha!, missed the line 83.